dask/dataframe/io/parquet/arrow.py

```diff
@@ -1030,7 +1045,7 @@ def _gather_metadata(
             paths = fs.sep.join([base, fns[0]])
 
             meta_path = fs.sep.join([paths, "_metadata"])
-            if fs.exists(meta_path):
+            if fs.exists(meta_path) and not ignore_metadata_file:
```

Comment:
What you have here is totally correct, but if we put the `ignore_metadata_file` condition first we might be able to avoid an unnecessary `fs.exists` call. This might not end up being important in practice, but I currently have S3 limits on the top of my mind : )

```suggestion
            if not ignore_metadata_file and fs.exists(meta_path):
```